packages/core/src/transport/httpRequest.ts

```diff
@@ -92,22 +89,30 @@ export function fetchKeepAliveStrategy(
   onResponse?: (r: HttpResponse) => void
 ) {
   const canUseKeepAlive = isKeepAliveSupported() && payload.bytesCount < bytesLimit
+
   if (canUseKeepAlive) {
-    const fetchUrl = endpointBuilder.build('fetch', payload)
-    fetch(fetchUrl, { method: 'POST', body: payload.data, keepalive: true, mode: 'cors' }).then(
-      monitor((response: Response) => onResponse?.({ status: response.status, type: response.type })),
-      monitor(() => {
-        const xhrUrl = endpointBuilder.build('xhr', payload)
-        // failed to queue the request
-        sendXHR(xhrUrl, payload.data, onResponse)
-      })
-    )
+    const fetchUrl = endpointBuilder.build('fetch-keepalive', payload)
+
+    fetch(fetchUrl, { method: 'POST', body: payload.data, keepalive: true, mode: 'cors' })
+      .then(monitor((response: Response) => onResponse?.({ status: response.status, type: response.type })))
+      .catch(monitor(() => fetchStrategy(endpointBuilder, payload, onResponse)))
   } else {
-    const xhrUrl = endpointBuilder.build('xhr', payload)
-    sendXHR(xhrUrl, payload.data, onResponse)
+    fetchStrategy(endpointBuilder, payload, onResponse)
   }
 }
 
+export function fetchStrategy(
+  endpointBuilder: EndpointBuilder,
+  payload: Payload,
+  onResponse?: (r: HttpResponse) => void
+) {
+  const fetchUrl = endpointBuilder.build('fetch', payload)
+
+  fetch(fetchUrl, { method: 'POST', body: payload.data, mode: 'cors' })
+    .then(monitor((response: Response) => onResponse?.({ status: response.status, type: response.type })))
+    .catch(() => onResponse?.({ status: 0 }))
```

Comment:
We do not monitor failed fetch request, as with the previous xhr implementation.

There is a lot of reason a request can fail, (no network, adblockers, ...). Filtering these reliably is difficult, and even if we could, monitoring the remaining errors does not give us actionable items